charts/httpd/templates/httpdRestart.yaml

```diff
@@ -0,0 +1,64 @@
+{{- if .Values.httpdRestart.enable }}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ include "httpd.fullname" . }}-restart-deployment
+  labels:
+{{ include "httpd.labels" . | indent 4 }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ include "httpd.fullname" . }}-restart-deployment
+  labels:
+{{ include "httpd.labels" . | indent 4 }}
+rules:
+  - apiGroups: ["apps", "extensions"]
+    resources: ["deployments"]
+    resourceNames: [{{ include "httpd.fullname" . }}]
+    verbs: ["get", "patch", "list", "watch"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: {{ include "httpd.fullname" . }}-restart-deployment
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: {{ include "httpd.fullname" . }}-restart-deployment
+subjects:
+  - kind: ServiceAccount
+    name: {{ include "httpd.fullname" . }}-restart-deployment
+---
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: {{ include "httpd.fullname" . }}-restart-cronjob
+  labels:
+{{ include "httpd.labels" . | indent 4 }}
+spec:
+  successfulJobsHistoryLimit: 1
+  failedJobsHistoryLimit: 2
+  concurrencyPolicy: Forbid
+  {{- with .Values.httpdRestart.cron }}
+  schedule: {{ toYaml . }}
+  {{- end }}
+  jobTemplate:
+    spec:
+      backoffLimit: 2
+      activeDeadlineSeconds: 600
```

Comment:
(nit) are these default values in Kubernetes? If no, what are the default (and are they sane for this use case)

Reply:
The `backoffLimit` is set by default to 6.
`activeDeadlineSeconds` as no default value.

see https://kubernetes.io/docs/concepts/workloads/controllers/job/#job-termination-and-cleanup

I feel inconfortable having a job with no limit, thus setting those 2 ... maybe we can leave the backoffLimit to it's 6 default and have a bigger deadline but a rollout restart should be fast or there is a problem.

Reply:
as we agreed I change to use the default backoffLimit of 6 and no activeDeadlineSeconds, will see later if we need to implement it.